blog-post/blog-post.md

```diff
@@ -0,0 +1,739 @@
+- empower library authors and support a community-driven process for language evolution
+- expand the capabilities of verification and analysis tooling
+
+In addition to growing and expanding the language, we will also be removing or reworking some
+existing features. We are already certain that we will be removing inheritance entirely. Additional
+changes are less certain, but we are considering potentially replacing or reworking features
+like try/catch, libraries, function pointers, type conversion, and reference semantics.
+
+We currently have a working prototype for Core Solidity. Most of the examples in this post typecheck
+and can produce executable code (although some make use of speculative syntax or unimplemented
+language constructs). Much of the core type theory is stable, but we want to add at least compile
+time evaluation and modules before we will consider the type system finalized. Extensive work
+remains to build out the standard library and reach feature parity with Classic Solidity.
+
+## A Note on Syntax
+
+Most of the work to date has been focused on the design and implementation of the type system, and
+associated code generation pipeline down to Yul. In order to avoid getting bogged down in
+bikeshedding around syntax, and with the desire to validate our core ideas with a working
+implementation as soon as possible, we moved ahead with a provisional syntax. You can expect
+extensive changes before release. Our current intention is to eventually match the syntax of Classic
+Solidity as much as possible. For any new syntax, you should expect that the final version will feel
+closer to languages like TypeScript or Rust than it does right now.
+
+## New Language Features
+
+Core Solidity takes ideas from pure functional programming languages (e.g. Haskell, Lean), as well
+as modern systems languages (e.g. Rust, Zig). We are extending Solidity with the following new
+features:
+
+- Algebraic datatypes (a.k.a sum / product types) and pattern matching
+- Generics / parametric polymorphism
+- Traits / typeclasses
+- Type inference
+- Higher order and anonymous functions
+- Compile time evaluation
+
+We think that taken together these core primitives will enable developers to produce stronger
+abstractions, write more modular and reusable code, and leverage the type system to enforce core
+safety properties.
+
+We will continue to support the kind of low level access to the EVM that is often required by
+production implementations: assembly will remain a core primitive, and we will extend it with the
+ability to directly call functions from the high level language in assembly blocks. Users will be
+able to disable the languages built in abstractions (e.g. contract dispatch generation, ABI
+decoding, default storage layout generation), following the "pay for what you use" philosophy of
+languages like Rust and C++.
+
+### Algebraic data types and pattern matching
```

Comment:
This section uses alternative and value constructer pretty interchangeably. I don't know if we should settle on one?